### PR TITLE
make npx-compatible version of playwright tests

### PR DESCRIPTION
**Pull request by Chiemezuo**
This is an experimental draft to use `playwright` for some a11y tests, keeping in mind that `playwright` is heavily dependent on node.js modules.

To run the dummy test, you would navigate to the `bokehjs` directory and run `npx playwright test /path/to/test/*.[spec|test].ts`

### CODE REVIEW

**Comment by Chiemezuo** [comment]
@mattpap 
What are your thoughts on this?

I tried running axe-core without playwright and it did work, however, it would be good to have a setup that would allow us to write more complex tests, and with an easier learning curve so that more work could be done on it by other people as well (in the future).

My next step from this would be to write a script that loads a Bokeh plot and performs some accessibility-related checks on it. However, I would appreciate your thoughts and possible gripes before I commit into that line of action.

**Comment by Chiemezuo** [comment]
Update: @mattpap replied me, but on Slack (because I asked for his feedback there as well) so I will share a paraphrased version of his response. 
He is fine with the progress of having some kind of testing setup so far that allows us run/create actual tests. When we will have an established baseline, it'll be easier to know exactly what these tests (and their integrations) should look like. For the `npx playwright test /path/to/*.[spec|test].ts` , it would be nice to have something that cooperates with Bokeh's existing tooling, but it's hard to say how practical or possible it might be for now. 

In summary: we can work with something that looks like this for now till we have something better.

File: bokehjs/package.json [hunk 1]
```diff
@@ -57,5 +57,9 @@
     "timezone": "^1.0.23",
     "tslib": "^2.6.3",
     "underscore.template": "^0.1.7"
+  },
+  "devDependencies": {
+    "@axe-core/playwright": "^4.10.0",
+    "@playwright/test": "^1.46.1"
```
Comment (mattpap) [inline bokehjs/package.json]: This needs to be moved to `bokehjs/test/package.json`.

File: bokehjs/test/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { defineConfig } from '@playwright/test';
```
Comment (mattpap) [inline bokehjs/test/playwright.config.ts:1]: You will have to create `tsconfig.json` file that matches settings of other classes of tests (with any relevant changes) and includes all `*.ts` files added in this PR. Then playwright has to be configured to use that setup somehow. This way we will have consistent type checking setup and it will be possible to lint this code. For linting, you will have to update `.eslintrc.js` accordingly (each `tsconfig.json` needs an entry there).

Comment (mattpap) [reply bokehjs/test/playwright.config.ts:1]: As to linting, when set up properly, you will get all the relevant errors to fix. A quick overview of changes need in this PR is:
1. No `;` unless needed to disambiguate code.
2. Trailing commas in all data structures and all other contexts.
3. Snake case instead of camel case or other, unless a name comes from an external source.
4. No unnecessary white space, e.g. in imports.

File: bokehjs/test/playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+export default defineConfig({
+  webServer: {
+    command: 'cd ../../docs/bokeh && make serve',
```
Comment (mattpap) [inline bokehjs/test/playwright.config.ts:4]: Documentation is non-trivial and slow to build, thus this will get old very quickly when running on a regular basis. Further more this makes bokehjs and in result its CI dependent on documentation, which is what we don't want. Ultimately this will have to be probably moved to bokeh's tests and a CI job set up that depends on documentation CI job, so that we don't have to build docs twice.

Comment (bryevdv) [reply bokehjs/test/playwright.config.ts:4]: I think ideally there is some `node make playright` or `pytest tests/playwright`  (or whatever) way to kick off these checks, that just assumes the docs are built, and then the checks only actually get run in the docs job (or one that depends on it, but probably simplest just to add a step to the existing job...)

**Comment by Chiemezuo** [comment]
Just as @tcmetzger suggested on Slack, I started a discussion (as a poll) on the topic of [what to do with axe-core](https://github.com/bokeh/bokeh/discussions/14057).
Please check it out if/when you can.

@bokeh/dev
